crates/sui-indexer-alt/src/handlers/obj_info.rs

```diff
@@ -93,6 +108,40 @@ impl Handler for ObjInfo {
             .execute(conn)
             .await?)
     }
+
+    async fn prune(&self, from: u64, to: u64, conn: &mut db::Connection<'_>) -> Result<usize> {
+        use sui_indexer_alt_schema::schema::obj_info::dsl;
+
+        let to_prune = self.pruning_lookup_table.take(from, to)?;
```

Comment:
seems sensible enough for pruner to take self. The handler will know best the shape of the data it needs to do any consistent pruning, would likely be unwieldy otherwise